kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionBuilderImpl.java

```diff
@@ -210,6 +211,20 @@ public Transaction build(Engine engine) {
       metadata = metadataWithIcebergCompatInfo.get();
     }
 
+    if (schema.isPresent() && !isNewTable) {
```

Comment:
Ping me when you are able to rebase off of https://github.com/delta-io/delta/pull/4267 and I'll take another look at this file (I think this will be the only one affected)